apps/src/gamelab/GameLabSprite.js

```diff
@@ -42,210 +41,220 @@ module.exports.createSprite = function (x, y, width, height) {
-    var yDelta = y - s.position.y;
-    var xDelta = x - s.position.x;
-    if (!isNaN(xDelta) && !isNaN(yDelta) && (xDelta !== 0 || yDelta !== 0)) {
-      var radiansAngle = Math.atan2(yDelta, xDelta);
-      s.rotation = 360 * radiansAngle / (2 * Math.PI);
-    }
-  };
-
-  // The scale value should include the horizontal stretch for animations.
-  s._getScaleX = function () {
-    return s.scale * s._horizontalStretch;
-  };
-
-  // The scale value should include the vertical stretch for animations.
-  s._getScaleY = function () {
-    return s.scale * s._verticalStretch;
-  };
-
-  /*
-   * @private
-   * For game lab, don't update the animation sizes because all frames are the same size.
-   */
-  s._syncAnimationSizes = function (animations, currentAnimation) {};
-
-  Object.defineProperty(s, 'frameDelay', {
-    enumerable: true,
-    get: function () {
-      if (s.animation) {
-        return s.animation.frameDelay;
-      }
-    },
-    set: function (value) {
-      if (s.animation) {
-        s.animation.frameDelay = value;
-      }
-    }
-  });
-
-  // Overriding these allows users to set a width for
+  // Overriding these allows users to set a width and height for
   // an animated sprite the same way they would an unanimated sprite.
   Object.defineProperty(s, 'width', {
     enumerable: true,
     configurable: true,
-    get: function () {
-      if (s._internalWidth === undefined) {
-        return 100;
-      } else if (s.animation) {
-        return s._internalWidth * s._horizontalStretch;
-      } else {
-        return s._internalWidth;
-      }
-    },
-    set: function (value) {
-      if (s.animation) {
-        s._horizontalStretch = value / s._internalWidth;
-      } else {
-        s._internalWidth = value;
-      }
-    }
+    get: getWidth,
+    set: setWidth
   });
-
-  // Overriding these allows users to set a height for
-  // an animated sprite the same way they would an unanimated sprite.
   Object.defineProperty(s, 'height', {
     enumerable: true,
     configurable: true,
-    get: function () {
-      if (s._internalHeight === undefined) {
-        return 100;
-      } else if (s.animation) {
-        return s._internalHeight * s._verticalStretch;
-      } else {
-        return s._internalHeight;
-      }
-    },
-    set: function (value) {
-      if (s.animation) {
-        s._verticalStretch = value / s._internalHeight;
-      } else {
-        s._internalHeight =  value;
-      }
-    }
+    get: getHeight,
+    set: setHeight
   });
 
-  // p5.play stores width unscaled, but users in
-  // Game Lab should have access to a scaled version.
-  s.getScaledWidth = function () {
-    return s.width * s.scale;
-  };
-
-  // p5.play stores height unscaled, but users in
-  // Game Lab should have access to a scaled version.
-  s.getScaledHeight = function () {
-    return s.height * s.scale;
-  };
+  // Attach our custom/override methods to the sprite
+  s.setAnimation = setAnimation.bind(s, this);
```

Comment:
In a few cases we still have to create a function for every sprite, such as `setAnimation` because we bind against a particular p5 instance, so we're not looking at any performance improvement from these.  Still, I think it's cleaner to put the implementation elsewhere and just do the binding here.